### PR TITLE
8332252: Clean up vmTestbase/vm/share

### PR DESCRIPTION
**Pull request by lmesnik**
The vmTestbase/vm/share is a shared test library for vmTestbase tests. This library contains a lot of code that is used by only by small number of tests or not used at all. There are no plans to actively develop new tests in vmTestsbase and improve this shared library. 
The final goal of this and the following PRs is to reduce the maintenance cost of vmTestbase by eliminating this library.

Also, this PR moves test-specific code into corresponding test directories to increase code locality. This allows later easier move tests from vmTestbase.

The few remaining classes include 
InMemoryJavaCompiler.java
that is very similar to same class from the standard testlibrary and could be merge with it and
ProcessUtils.java
which is used by
test/hotspot/jtreg/runtime/Thread/TestBreakSignalThreadDump.java
and thus should be moved into the standard testlibrary.
The stack and options might be merged in nsk/share test library.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8332252](https://bugs.openjdk.org/browse/JDK-8332252): Clean up vmTestbase/vm/share (**Enhancement** - P4)

### Reviewers
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19727/head:pull/19727` \
`$ git checkout pull/19727`

Update a local copy of the PR: \
`$ git checkout pull/19727` \
`$ git pull https://git.openjdk.org/jdk.git pull/19727/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19727`

View PR using the GUI difftool: \
`$ git pr show -t 19727`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19727.diff">https://git.openjdk.org/jdk/pull/19727.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19727#issuecomment-2168407566)

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/vm/compiler/complog/share/ProcessExecutor.java [hunk 1]
```diff
@@ -138,8 +131,8 @@ public int waitFor(long timeout) {
         return -1;
     }
 
-    public int getPid() {
-        return ProcessUtils.getPid(process);
+    public long getPid() {
+        return process.toHandle().pid();
```
Comment (plummercj) [inline test/hotspot/jtreg/vmTestbase/vm/compiler/complog/share/ProcessExecutor.java:135]: `tohandle()` is not necessary.

File: test/hotspot/jtreg/vmTestbase/metaspace/share/TriggerUnloadingWithWhiteBox.java [hunk 2]
```diff
@@ -20,7 +20,7 @@
  * or visit www.oracle.com if you need additional information or have any
  * questions.
  */
-package vm.share.gc;
+package metaspace.share;
```
Comment (coleenp) [inline test/hotspot/jtreg/vmTestbase/metaspace/share/TriggerUnloadingWithWhiteBox.java:23]: There's a triggerUnloading call here:

test/lib/jdk/test/lib/classloader/ClassUnloadCommon.java

You might be able to also remove this file (and maybe the others) and use the ClassUnloadCommon version.

**Review by coleenp — COMMENTED** [review]
Thank you so much for this cleanup work.  I have one suggestion which hopefully will work.

File: test/hotspot/jtreg/vmTestbase/vm/compiler/complog/share/LogCompilationTest.java [hunk 3]
```diff
@@ -29,7 +29,6 @@
 import nsk.share.log.LogSupport;
 import vm.share.options.Option;
 import vm.share.options.OptionSupport;
-import vm.share.process.ProcessExecutor;
```
Comment (plummercj) [inline test/hotspot/jtreg/vmTestbase/vm/compiler/complog/share/LogCompilationTest.java:32]: You got rid of this import, but ProcessExecutor is still referenced below. Is this file even referenced during test execution?

Comment (lmesnik) [reply test/hotspot/jtreg/vmTestbase/vm/compiler/complog/share/LogCompilationTest.java:32]: The ProcessExecutor has been moved into this package, so it is local package now. 
Double checked that it is used and tests
 jtreg:open/test/hotspot/jtreg/vmTestbase/vm/compiler/complog
still pass.

File: test/hotspot/jtreg/vmTestbase/metaspace/share/TriggerUnloadingWithWhiteBox.java (hunk 2 above)
Comment (lmesnik) [reply test/hotspot/jtreg/vmTestbase/metaspace/share/TriggerUnloadingWithWhiteBox.java:23]: Thanks. I filed 
https://bugs.openjdk.org/browse/JDK-8334320
The functionality is little different so more testing might be required for changed tests.

**Review by plummercj — APPROVED** [review]

**Review by stefank — COMMENTED** [review]
Some of the moves/renames messed up the sort-order of the imports. Could you take a pass over the patch and clean that up?

**Review by plummercj — APPROVED** [review]

**Pull request closed without merging**